src/io.c

```diff
@@ -85,3 +83,14 @@ chomp(char *string, ssize_t length)
 		string[length - 1] = '\0';
 	}
 }
+
+static char *
+eager_strpbrk(const char *string, const char *separators) {
```

Comment:
Given that `$IFS` is actually a list of possible field separating characters, I think `separators` is a better name for what it is. In case you weren’t aware of this behavior, this is exactly how `strtok_r` works too, so there should be no regression. `strtok_r` [uses](https://github.molgen.mpg.de/git-mirror/glibc/blob/master/string/strtok_r.c#L63) `strpbrk` internally, while `strpbrk` is [implemented via two loops](https://github.molgen.mpg.de/git-mirror/glibc/blob/master/string/strpbrk.c) with the outer loop iterating a given string char-by-char and the inner one checking these characters for inclusion in separators.

Default `$IFS` set by Bash includes all regular whitespace characters (space, tab and newline):

``` bash
$ echo "$IFS" | cat -te
 ^I$
$
```